src/compiler/checker.ts

```diff
@@ -12744,6 +12744,10 @@ namespace ts {
                 if (declaration.type) {
                     return getTypeFromTypeNode(declaration.type);
                 }
+                const jsDocType = isInJavaScriptFile(declaration) && getTypeForDeclarationFromJSDocComment(declaration);

```

Comment:
Nit: the style here differs from the case immediately below it, although the two seem to be accomplishing the same thing. I personally prefer the below style (though with `if (type) return type;` on one line).